gateway/ln-gateway/src/cln.rs

```diff
@@ -58,18 +59,40 @@ pub struct HtlcAccepted {
 
 pub struct ClnRpcRef {
     // CLN RPC client
-    pub ln_rpc: Arc<Mutex<ClnRpc>>,
+    pub ln_rpc: Arc<ClnRpc>,
     // Config directory
     pub work_dir: PathBuf,
 }
 
+/// Core Lightning RPC client
+#[derive(Debug, Clone)]
+pub struct ClnRpc {
+    socket: PathBuf,
+}
+
+impl ClnRpc {
+    pub fn new(socket: PathBuf) -> ClnRpc {
+        ClnRpc { socket }
+    }
+
+    /// Creates a new RPC client for a request.
+    ///
+    /// This operation is cheap enough to do it for each request since it merely connects to a UNIX
+    /// domain socket without doing any further initialization.
+    async fn rpc_client(&self) -> Result<cln_rpc::ClnRpc, LightningError> {
+        cln_rpc::ClnRpc::new(&self.socket)
+            .await
+            .map_err(|_e| LightningError(None))
```

Comment:
If we are going to swallow that error, I would rather log it. Would be even better to return it somehow. Can't we return `anyhow::Result` here anyway?

Reply:
Good catch! I missed that one when tidying up the original change. I went for the logging approach since changing the error type would require far-reaching API changes in parts that are likely to be changed in the GW refactoring effort anyway. 

@okjodom: What's our error strategy for the new gateway? Can we have more expressive errors while still supporting LN error codes?

Reply:
Well define strict Ln error codes over grpc spec.

Logging within gatewayd is more open, should conform to logging pattern we define everywhere else